R/plumber-step.R

```diff
@@ -120,26 +136,60 @@ PlumberEndpoint <- R6Class(
   "PlumberEndpoint",
   inherit = PlumberStep,
   public = list(
+    #' @field verbs a character vector. http methods. For historical reasons we have
+    #' to accept multiple verbs for a single path. Now it's simpler to just parse
+    #' each separate verb/path into its own endpoint, so we just do that.
     verbs = NA,
+    #' @field path a character string. endpoint path
     path = NA,
+    #' @field comments endpoint comments
     comments = NA,
+    #' @field responses endpoint responses
     responses = NA,
+    #' @description retrieve endpoint typed parameters
     getTypedParams = function(){
-      data.frame(name=private$regex$names, type=private$regex$types, stringsAsFactors = FALSE)
+      data.frame(name = private$regex$names,
+                 type = private$regex$types,
+                 serialization = private$regex$serializations,
+                 stringsAsFactors = FALSE)
     },
+    #' @field params endpoint parameters
     params = NA,
+    #' @field tags endpoint tags
     tags = NA,
-    canServe = function(req){
-      req$REQUEST_METHOD %in% self$verbs && !is.na(stringi::stri_match(req$PATH_INFO, regex=private$regex$regex)[1,1])
+    #' @description ability to serve request
+    #' @param req a request object
+    #' @param res a response object
+    #' @return a logical. `TRUE` when endpoint can serve request.
+    canServe = function(req, res){
+      if (!is.na(stri_match_first_regex(req$PATH_INFO, private$regex$regex)[1,1])) {
+        if (req$REQUEST_METHOD %in% self$verbs) {
+          return(TRUE)
+        } else {
+          res$status = 405L
+          res$setHeader("Allow", paste(self$verbs, collapse = ", "))
```

Comment:
This logic have to be done after all routes have failed to serve.

Ex:
```
/home POST
/home GET
```

This would end up producing an `Allow` header on a GET post.